### ISSUE TITLE
Implement proper way of doing early stopping

### ISSUE THREAD

**Issue by jabalazs** [jabalazs/implicit_emotion#9]
The idea would be to control the stopping policies from a centralized place, so we could for example tell the training procedure to stop if any of the following conditions are met:

Epoch greater than 10
Learning rate lower than 1e-5
Validation accuracy has not improved in the last 2 epochs
